### DISCUSSION TITLE
Add annotation to force recreation of Jobs

### DISCUSSION THREAD

**Discussion by tun0** [fluxcd/flux2#3433] [Proposals] [answered]
I recently ran into [this](https://fluxcd.io/flux/use-cases/running-jobs/) nice article. Which got me thinking: you might not really need the dependency stuff, but do would like for a Job to be kicked off when a new image is available. In which case the `Kustomization` part feels rather overkill. A "cleaner" solution would be to annotate said Job(s) with say `kustomize.toolkit.fluxcd.io/force: true` and have those Jobs recreated whenever an immutable field changes.

See also [slack](https://cloud-native.slack.com/archives/CLAJ40HV3/p1671619908544919).

**Comment by stefanprodan** [comment, marked as answer]
Update: This has been released in Flux 0.39.0

I've started implementing this here: https://github.com/fluxcd/pkg/pull/453. After the SSA changes are merged and released, I'll enable the selector in kustomize-controller. We are not using boolean values for labels and annotations, the key will be `kustomize.toolkit.fluxcd.io/force` with `enabled` and `disabled` as values.

**Reply by tun0** [reply]
Awesome!

**Reply by guicholeo** [reply]
@stefanprodan can this be applied to the Helm Release as well? I'm deploying a Job Helm Chart via a Helm Release.

**Reply by tun0** [reply]
Pretty sure that helm itself offers quite some methods for triggering Job executions?

**Reply by guicholeo** [reply]
They do, I want to recreate the Helm Release to pass the values I'm using in it. And the Helm Release is a dependency to other HRs

**Reply by stefanprodan** [reply]
@guicholeo there is no need for such an annotation inside charts, just set `force` in the HelmRelease upgrade spec.
